drivers/soundwire/cadence_master.c

```diff
@@ -1375,6 +1375,12 @@ int sdw_cdns_clock_restart(struct sdw_cdns *cdns, bool bus_reset)
 
 	cdns_update_config(cdns);
 
+	ret = sdw_cdns_enable_interrupt(cdns, true);
```

Comment:
@plbossart  the issue “rt715 sdw:3:25d:715:0: Initialization not complete, timed out” should be caused by this change.
the normal status should be
````
[  207.838359] intel-sdw sdw-master-3: intel_resume_runtime start
[  207.838361] intel-sdw sdw-master-3: intel_resume_runtime resume from BUS_RESET
[  207.838974] rt715 sdw:3:25d:715:0: sdw_modify_slave_status: changing status slave 1 status 1 new status 0
[  207.838975] rt715 sdw:3:25d:715:0: sdw_modify_slave_status: initializing completion for Slave 1
[  207.838976] intel-sdw sdw-master-3: sdw_cdns_clock_restart: start
[  207.839217] intel-sdw sdw-master-3: sdw_cdns_clock_restart: end
[  207.839218] intel-sdw sdw-master-3: intel_resume_runtime done
[  207.840113] intel-sdw sdw-master-3: Slave status change
[  207.840118] intel-sdw sdw-master-3: Slave 0 ATTACHED
[  207.840119] intel-sdw sdw-master-3: sdw_handle_slave_status: start
[  207.840120] intel-sdw sdw-master-3: Slave attached, programming device number
[  207.840295] intel-sdw sdw-master-3: SDW Slave Addr: 21025d071500
[  207.840296] intel-sdw sdw-master-3: SDW Slave class_id 0, part_id 715, mfg_id 25d, unique_id 1, version 2
````
The abnormal case is:
````
[  218.431430] intel-sdw sdw-master-3: intel_resume_runtime start
[  218.431434] intel-sdw sdw-master-3: intel_resume_runtime resume from BUS_RESET
[  218.432009] rt715 sdw:3:25d:715:0: sdw_modify_slave_status: changing status slave 1 status 1 new status 0
[  218.432011] rt715 sdw:3:25d:715:0: sdw_modify_slave_status: initializing completion for Slave 1
[  218.432015] intel-sdw sdw-master-3: sdw_cdns_clock_restart: start
[  218.432036] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  218.432088] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[  218.434677] intel-sdw sdw-master-3: sdw_cdns_clock_restart: end
[  218.434680] intel-sdw sdw-master-3: intel_resume_runtime done
[  218.932260] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[  218.932343] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[  220.442971] rt715 sdw:3:25d:715:0: Initialization not complete, timed out
````

I guess: there is a interrupt triggered immediately after bus reset but before interrupt is enabled. So the the interrupt for attachment is missed. I tested on CML-H. It is easy to reproduce this issue with this change, and the issue disappeared after I moved sdw_cdns_enable_interrupt ahead of bus reset.

Reply:
@RanderWang ok, I guess that makes sense, it's indeed risky if there is an interrupt generated but the interrupts are not enabled.
However we may want to make sure all interrupts status are cleared before re-enabling them prior to a bus reset?

Reply:
@plbossart   yes, it is a defensive programming. please check https://github.com/thesofproject/linux/pull/1671